app/controllers/artefacts_controller.rb

```diff
@@ -42,7 +42,9 @@ def history
   end
 
   def archive
-
+    if @artefact.state == 'archived'
```

Comment:
`Artefact` has an [`#archived?` method](https://github.com/alphagov/govuk_content_models/blob/master/app/models/artefact.rb#L353-L355), which would be better to use here, eg:

```ruby
if @artefact.archived?
  …
end
```